bench/rttanalysis: TestBenchmarkExpectation failed [timeout]

bench/rttanalysis.TestBenchmarkExpectation [failed](https://mesolite.cluster.engflow.com/invocations/default/c4558e64-3bc6-4a8c-a360-21efdc7fb749?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvYmVuY2gvcnR0YW5hbHlzaXM6cnR0YW5hbHlzaXNfdGVzdA==) on master @ [352b1d2522799537674493e1b4775b259ab0c725](https://github.com/cockroachdb/cockroach/commits/352b1d2522799537674493e1b4775b259ab0c725):


```
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 937324
	external/org_golang_x_sync/errgroup/errgroup.go:75 +0x96
=== RUN   TestBenchmarkExpectation/Truncate/truncate_1_column_2_rows
		TestBenchmarkExpectation/Truncate/truncate_1_column_2_rows (15m0s)
=== RUN   TestBenchmarkExpectation/Truncate/truncate_2_column_1_rows
		TestBenchmarkExpectation/Truncate/truncate_2_column_1_rows (15m0s)
=== RUN   TestBenchmarkExpectation/AlterTableAddForeignKey/alter_table_add_1_foreign_key
		TestBenchmarkExpectation/AlterTableAddForeignKey/alter_table_add_1_foreign_key (15m0s)
=== RUN   TestBenchmarkExpectation/AlterTableAddForeignKey/alter_table_add_2_foreign_keys
		TestBenchmarkExpectation/AlterTableAddForeignKey/alter_table_add_2_foreign_keys (15m0s)
=== RUN   TestBenchmarkExpectation/AlterTableDropConstraint/alter_table_drop_1_check_constraint
		TestBenchmarkExpectation/AlterTableDropConstraint/alter_table_drop_1_check_constraint (15m0s)
=== RUN   TestBenchmarkExpectation/AlterTableAddForeignKey/alter_table_add_3_foreign_keys
		TestBenchmarkExpectation/AlterTableAddForeignKey/alter_table_add_3_foreign_keys (15m0s)
=== RUN   TestBenchmarkExpectation/SystemDatabaseQueries/select_system.users_without_schema_Name
		TestBenchmarkExpectation/SystemDatabaseQueries/select_system.users_without_schema_Name (15m0s)
=== RUN   TestBenchmarkExpectation/AlterTableAddCheckConstraint/alter_table_add_2_check_constraints
		TestBenchmarkExpectation/AlterTableAddCheckConstraint/alter_table_add_2_check_constraints (15m0s)
=== RUN   TestBenchmarkExpectation/AlterTableConfigureZone/alter_table_configure_zone_7_replicas_
		TestBenchmarkExpectation/AlterTableConfigureZone/alter_table_configure_zone_7_replicas_ (15m0s)
=== RUN   TestBenchmarkExpectation/GenerateObjects/generate_50000_tables
		TestBenchmarkExpectation/GenerateObjects/generate_50000_tables (15m0s)
=== RUN   TestBenchmarkExpectation/GenerateObjects/generate_1000_tables_-_this_test_should_use_the_same_number_of_RTTs_as_for_10_tables
		TestBenchmarkExpectation/GenerateObjects/generate_1000_tables_-_this_test_should_use_the_same_number_of_RTTs_as_for_10_tables (15m0s)
=== RUN   TestBenchmarkExpectation/AlterTableAddForeignKey/alter_table_add_foreign_key_with_3_columns
		TestBenchmarkExpectation/AlterTableAddForeignKey/alter_table_add_foreign_key_with_3_columns (15m0s)
=== RUN   TestBenchmarkExpectation/AlterTableSplit/alter_table_split_at_2_values
		TestBenchmarkExpectation/AlterTableSplit/alter_table_split_at_2_values (15m0s)
=== RUN   TestBenchmarkExpectation/SystemDatabaseQueries
		TestBenchmarkExpectation/SystemDatabaseQueries (15m0s)
=== RUN   TestBenchmarkExpectation/Truncate/truncate_1_column_1_row
		TestBenchmarkExpectation/Truncate/truncate_1_column_1_row (15m0s)
=== RUN   TestBenchmarkExpectation/AlterTableSplit
		TestBenchmarkExpectation/AlterTableSplit (15m0s)
=== RUN   TestBenchmarkExpectation/DropSequence
		TestBenchmarkExpectation/DropSequence (15m0s)
=== RUN   TestBenchmarkExpectation/GenerateObjects/generate_100_tables_from_template
		TestBenchmarkExpectation/GenerateObjects/generate_100_tables_from_template (15m0s)
=== RUN   TestBenchmarkExpectation/DropView/drop_1_view
		TestBenchmarkExpectation/DropView/drop_1_view (15m0s)
=== RUN   TestBenchmarkExpectation/SystemDatabaseQueries/select_system.users_with_schema_Name
		TestBenchmarkExpectation/SystemDatabaseQueries/select_system.users_with_schema_Name (15m0s)
=== RUN   TestBenchmarkExpectation/Truncate
		TestBenchmarkExpectation/Truncate (15m0s)
=== RUN   TestBenchmarkExpectation/AlterTableAddForeignKey
		TestBenchmarkExpectation/AlterTableAddForeignKey (15m0s)
=== RUN   TestBenchmarkExpectation/AlterTableConfigureZone
		TestBenchmarkExpectation/AlterTableConfigureZone (15m0s)
=== RUN   TestBenchmarkExpectation/DropSequence/drop_1_sequence
		TestBenchmarkExpectation/DropSequence/drop_1_sequence (15m0s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126750 bench/rttanalysis: TestBenchmarkExpectation failed [Jobs/jobs_page_type_filtered_no_matches] [A-jobs C-test-failure O-robot P-3 T-jobs branch-release-24.1]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBenchmarkExpectation.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-51588

The test hit a 15-minute timeout.

```
=== NAME  TestBenchmarkExpectation/DropView
panic: test timed out after 15m0s
    running tests:
        TestBenchmarkExpectation (15m0s)
```

This goroutine seems related to schema change. Looks like it's waiting in `WaitForOneVersion`:
```
goroutine 940644 [chan receive]:
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).divideAndSendBatchToRanges.func1()
    pkg/kv/kvclient/kvcoord/dist_sender.go:1858 +0x11d
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).divideAndSendBatchToRanges(0xc007008008, {0x6571668, 0xc001de17c0}, 0xc0022d7a40, {{0xc00790fd48, 0x4, 0x8}, {0xc00790fdc0, 0x4, 0x4}}, ...)
    pkg/kv/kvclient/kvcoord/dist_sender.go:2056 +0x14c8
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).Send(0xc007008008, {0x6571668, 0xc001de1720}, 0xc0022d7a40)
    pkg/kv/kvclient/kvcoord/dist_sender.go:1269 +0x518
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnLockGatekeeper).SendLocked(0xc0029fa230, {0x6571668, 0xc001de1720}, 0xc0022d7a40)
    pkg/kv/kvclient/kvcoord/txn_lock_gatekeeper.go:77 +0x1b6
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnMetricRecorder).SendLocked(0xc0029fa1e0, {0x6571668?, 0xc001de1720?}, 0xc0022d7a40?)
    pkg/kv/kvclient/kvcoord/txn_interceptor_metric_recorder.go:41 +0xd0
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnSpanRefresher).sendLockedWithRefreshAttempts(0xc0029fa100, {0x6571668, 0xc001de1720}, 0xc0022d7a40, 0x5)
    pkg/kv/kvclient/kvcoord/txn_interceptor_span_refresher.go:228 +0x45
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnSpanRefresher).SendLocked(0xc0029fa100, {0x6571668, 0xc001de1720}, 0xc00c1eb280?)
    pkg/kv/kvclient/kvcoord/txn_interceptor_span_refresher.go:162 +0xb3
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnCommitter).SendLocked(0xc0029fa0c0, {0x6571668, 0xc001de1720}, 0xc0022d7a40)
    pkg/kv/kvclient/kvcoord/txn_interceptor_committer.go:144 +0x49c
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnPipeliner).SendLocked(0xc0029f9f68, {0x6571668, 0xc001de1720}, 0xc00a5eb260?)
    pkg/kv/kvclient/kvcoord/txn_interceptor_pipeliner.go:334 +0x20a
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnWriteBuffer).SendLocked(0xc0016ddbd0?, {0x6571668?, 0xc001de1720?}, 0x5?)
    pkg/kv/kvclient/kvcoord/txn_interceptor_write_buffer.go:215 +0x555
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnSeqNumAllocator).SendLocked(0xc0029f9ea0, {0x6571668, 0xc001de1720}, 0xc0022d7a40)
    pkg/kv/kvclient/kvcoord/txn_interceptor_seq_num_allocator.go:112 +0x259
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnHeartbeater).SendLocked(0xc0029f9de8, {0x6571668, 0xc001de1720}, 0xc0022d7a40)
    pkg/kv/kvclient/kvcoord/txn_interceptor_heartbeater.go:265 +0x48a
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*TxnCoordSender).Send(0xc0029f9c08, {0x6571668, 0xc00a63d900}, 0xc0022d7a40)
    pkg/kv/kvclient/kvcoord/txn_coord_sender.go:558 +0x5b1
github.com/cockroachdb/cockroach/pkg/kv.(*DB).sendUsingSender(0xc00844dd60, {0x6571668, 0xc00a63d900}, 0xc0022d7a40, {0x7fd86cfdc830, 0xc0029f9c08})
    pkg/kv/db.go:1149 +0xe7
github.com/cockroachdb/cockroach/pkg/kv.(*Txn).Send(0xc0041a6fd0, {0x6571668, 0xc00a63d900}, 0xc0022d7a40)
    pkg/kv/txn.go:1368 +0x265
github.com/cockroachdb/cockroach/pkg/kv.sendAndFill({0x6571668, 0xc00a63d900}, 0xc00f13fba8, 0xc002b38608)
    pkg/kv/db.go:941 +0x10a
github.com/cockroachdb/cockroach/pkg/kv.(*Txn).Run(0xc0041a6fd0, {0x6571668, 0xc00a63d900}, 0xc002b38608)
    pkg/kv/txn.go:860 +0x65
github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/catkv.catalogQuery.query({{{0xa06ed00, 0xa06ed20, {0x1}}, {0xa06ed00}, 0x0}, 0x0, {0x5971e74, 0x3}}, {0x6571668, 0xc00a63d900}, ...)
    pkg/sql/catalog/internal/catkv/catalog_query.go:55 +0x17a
github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/catkv.catalogReader.GetByIDs({{{0xa06ed00, 0xa06ed20, {0x1}}, {0xa06ed00}, 0x0}}, {0x6571668?, 0xc00a63d900?}, 0x44d88d?, {0xc00790fd38, 0x1, ...}, ...)
    pkg/sql/catalog/internal/catkv/catalog_reader.go:356 +0x1d3
github.com/cockroachdb/cockroach/pkg/sql/catalog/internal/catkv.(*cachedCatalogReader).GetByIDs(0xc02f398d20, {0x6571668, 0xc00a63d900}, 0xc0041a6fd0, {0xc00790fd38, 0x1, 0x1}, 0x0, {0x5971e74, 0x3})
    pkg/sql/catalog/internal/catkv/catalog_reader_cached.go:357 +0x203
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*Manager).maybeGetDescriptorsWithoutValidation.func1({0x6571668, 0xc00a63d900}, 0xc0041a6fd0)
    pkg/sql/catalog/lease/lease.go:152 +0x10b
github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec(0xc0041a6fd0, {0x6571668, 0xc00a63d900}, 0xc00f1405c8)
    pkg/kv/txn.go:1131 +0x2b1
github.com/cockroachdb/cockroach/pkg/kv.runTxn({0x6571668, 0xc00a63d900}, 0xc0041a6fd0, 0x1?)
    pkg/kv/db.go:1074 +0x3e
github.com/cockroachdb/cockroach/pkg/kv.(*DB).TxnWithAdmissionControl(0x5418da0?, {0x6571668, 0xc00a63d900}, 0xf140600?, 0xc0?, 0x0, 0xc00f1405c8)
    pkg/kv/db.go:1037 +0xa5
github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn(...)
    pkg/kv/db.go:1012
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*Manager).maybeGetDescriptorsWithoutValidation(0xc005fbf908, {0x6571668, 0xc00a63d900}, {0xc00790fd38, 0x1, 0x1}, 0x1)
    pkg/sql/catalog/lease/lease.go:149 +0xf9
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*Manager).WaitForOneVersion(0xc005fbf908, {0x6571668, 0xc00a63d900}, 0x21, {0x654c2a0, 0xc0268a32a0}, {0xf4240, 0x3b9aca00, 0x3ff8000000000000, 0x0, ...})
    pkg/sql/catalog/lease/lease.go:466 +0x350
github.com/cockroachdb/cockroach/pkg/sql.WaitToUpdateLeases({0x6571668, 0xc00a63d900}, 0xc005fbf908, {0x654c2a0, 0xc0268a32a0}, 0x21)
    pkg/sql/schema_changer.go:1488 +0x13f
github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChanger).exec.func1(0x0)
    pkg/sql/schema_changer.go:842 +0xa5
github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChanger).exec(0xc003b0f290, {0x656fe58, 0xc00f4d3d40})
    pkg/sql/schema_changer.go:959 +0xc95
github.com/cockroachdb/cockroach/pkg/sql.schemaChangeResumer.Resume.func1(0x21, 0x0, 0x0, {0x0, 0x0, 0x0?}, {0x0, 0x0, 0x5663f3?})
    pkg/sql/schema_changer.go:2904 +0x785
github.com/cockroachdb/cockroach/pkg/sql.schemaChangeResumer.Resume({0x260b3db?}, {0x656fe58, 0xc00f4d3d40}, {0x5674ca0?, 0xc0025e4b88})
    pkg/sql/schema_changer.go:3060 +0x91a
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0xc001efa480?, 0x0?, 0xc00f141928, {0x654c3f0?, 0xc0025e44c0?}, {0x656fe58?, 0xc00f4d3d40?}, {0x5674ca0?, 0xc0025e4b88?})
    pkg/jobs/registry.go:1717 +0xb2
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0xc002b86008, {0x656fe20, 0xc00d2bcff0}, {0x5674ca0, 0xc0025e4b88}, {0x654c3f0, 0xc0025e44c0}, 0xc001efa480, {0xc003f20d40, 0x7}, ...)
    pkg/jobs/registry.go:1718 +0xba5
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0xc002b86008, {0x656fe20, 0xc00d2bcff0}, {0x654c3f0, 0xc0025e44c0}, 0xc001efa480, {0xc003f20d40, 0x7}, {0xc008d08450, 0x17})
    pkg/jobs/adopt.go:410 +0x545
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func1({0x6571668?, 0xc009736be0?})
    pkg/jobs/adopt.go:263 +0x13c
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0x6571668, 0xc009736be0})
    pkg/util/stop/stopper.go:438 +0x8a
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 940603
    pkg/util/stop/stopper.go:436 +0x137
```

bench/rttanalysis.TestBenchmarkExpectation [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20283498?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20283498?buildTab=artifacts#/) on master @ [53ce6cb794977428d73db4706d43c70f5563e0cb](https://github.com/cockroachdb/cockroach/commits/53ce6cb794977428d73db4706d43c70f5563e0cb):


```
    validate_benchmark_data.go:120: success: expected ORMQueries/information_schema._pg_index_position to perform KV lookups in [4, 4], got 4
    validate_benchmark_data.go:120: success: expected ORMQueries/django_comment_introspection_with_comments to perform KV lookups in [6, 6], got 5
    validate_benchmark_data.go:120: success: expected ORMQueries/django_table_introspection_8_tables to perform KV lookups in [6, 6], got 5
    validate_benchmark_data.go:120: success: expected Jobs/jobs_page_type_filtered to perform KV lookups in [3, 3], got 3
    validate_benchmark_data.go:120: success: expected Jobs/jobs_page_latest_50 to perform KV lookups in [3, 3], got 3
    validate_benchmark_data.go:117: fail: expected Jobs/crdb_internal.system_jobs to perform KV lookups in [3, 3], got 6
    validate_benchmark_data.go:120: success: expected ORMQueries/hasura_column_descriptions_modified to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:120: success: expected Grant/grant_all_on_1_table to perform KV lookups in [8, 8], got 8
    validate_benchmark_data.go:120: success: expected Grant/grant_all_on_2_tables to perform KV lookups in [8, 8], got 8
    validate_benchmark_data.go:120: success: expected DropView/drop_1_view to perform KV lookups in [14, 14], got 14
    validate_benchmark_data.go:120: success: expected ORMQueries/npgsql_fields to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:120: success: expected ORMQueries/npgsql_types to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:120: success: expected DropView/drop_2_views to perform KV lookups in [15, 15], got 15
    validate_benchmark_data.go:120: success: expected DropTable/drop_1_table to perform KV lookups in [13, 13], got 13
    validate_benchmark_data.go:120: success: expected DropTable/drop_3_tables to perform KV lookups in [17, 17], got 17
    validate_benchmark_data.go:120: success: expected ORMQueries/asyncpg_types to perform KV lookups in [4, 4], got 4
    validate_benchmark_data.go:120: success: expected DropSequence/drop_3_sequences to perform KV lookups in [17, 17], got 17
    validate_benchmark_data.go:120: success: expected DropTable/drop_2_tables to perform KV lookups in [15, 15], got 15
    validate_benchmark_data.go:120: success: expected DropSequence/drop_1_sequence to perform KV lookups in [13, 13], got 13
    validate_benchmark_data.go:120: success: expected DropSequence/drop_2_sequences to perform KV lookups in [15, 15], got 15
    validate_benchmark_data.go:120: success: expected AlterTableAddCheckConstraint/alter_table_add_1_check_constraint to perform KV lookups in [13, 13], got 13
    validate_benchmark_data.go:120: success: expected ORMQueries/prisma_column_descriptions_updated to perform KV lookups in [3, 3], got 4
    validate_benchmark_data.go:120: success: expected AlterTableAddCheckConstraint/alter_table_add_3_check_constraints to perform KV lookups in [13, 13], got 13
    validate_benchmark_data.go:120: success: expected AlterTableAddCheckConstraint/alter_table_add_2_check_constraints to perform KV lookups in [13, 13], got 13
    validate_benchmark_data.go:120: success: expected DropDatabase/drop_database_0_tables to perform KV lookups in [15, 15], got 15
    validate_benchmark_data.go:120: success: expected ORMQueries/prisma_column_descriptions to perform KV lookups in [6, 6], got 6
    validate_benchmark_data.go:120: success: expected VirtualTableQueries/show_create_all_routines to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:120: success: expected DropView/drop_3_views to perform KV lookups in [15, 15], got 15
    validate_benchmark_data.go:120: success: expected ORMQueries/liquibase_migrations to perform KV lookups in [4, 4], got 4
    validate_benchmark_data.go:120: success: expected UseManyRoles/use_2_roles to perform KV lookups in [2, 2], got 2
    validate_benchmark_data.go:120: success: expected ORMQueries/column_descriptions_json_agg to perform KV lookups in [6, 6], got 6
    validate_benchmark_data.go:120: success: expected ORMQueries/prisma_types_4 to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:120: success: expected Jobs/non_admin_crdb_internal.system_jobs to perform KV lookups in [2, 2], got 2
    validate_benchmark_data.go:120: success: expected Jobs/non_admin_show_jobs to perform KV lookups in [2, 2], got 2
    validate_benchmark_data.go:120: success: expected ORMQueries/prisma_types_16 to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:120: success: expected UseManyRoles/use_50_roles to perform KV lookups in [2, 2], got 2
    validate_benchmark_data.go:120: success: expected VirtualTableQueries/show_create_all_triggers to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:120: success: expected ORMQueries/liquibase_migrations_on_multiple_dbs to perform KV lookups in [8, 8], got 8
    validate_benchmark_data.go:73: see the -rewrite flag to re-run the benchmarks and adjust the expectations
    validate_benchmark_data.go:74: usage: ./dev test --rewrite pkg/bench/rttanalysis -f=TestBenchmarkExpectation/ --test-args '-rewrite-iterations=N'
    validate_benchmark_data.go:107: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b03b0378e25aebc8fcd368039859e7a0/logTestBenchmarkExpectation1863931570
--- FAIL: TestBenchmarkExpectation (224.94s)
--- FAIL: TestBenchmarkExpectation (225.35s)
=== RUN   TestBenchmarkExpectation/Jobs
--- FAIL: TestBenchmarkExpectation/Jobs (152.16s)
=== RUN   TestBenchmarkExpectation/Jobs/crdb_internal.system_jobs
    rtt_analysis_bench.go:224: Jobs/crdb_internal.system_jobs: got 6, expected 3
    rtt_analysis_bench.go:230: wrote jaeger trace to /artifacts/tmp/_tmp/b03b0378e25aebc8fcd368039859e7a0/logTestBenchmarkExpectation1863931570/Jobs_crdb_internal.system_jobs.jaeger.json
--- FAIL: TestBenchmarkExpectation/Jobs/crdb_internal.system_jobs (120.18s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126750 bench/rttanalysis: TestBenchmarkExpectation failed [Jobs/jobs_page_type_filtered_no_matches] [A-jobs C-test-failure O-robot P-3 T-jobs branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBenchmarkExpectation.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/bench/rttanalysis/BUILD.bazel
+++ b/pkg/bench/rttanalysis/BUILD.bazel
@@ -13,6 +13,8 @@ go_library(
     visibility = ["//visibility:public"],
     deps = [
         "//pkg/base",
+        "//pkg/jobs",
+        "//pkg/jobs/jobspb",
         "//pkg/kv/kvclient/kvcoord",
         "//pkg/sql",
         "//pkg/sql/parser",
@@ -56,9 +58,9 @@ go_test(
     data = glob(["testdata/**"]),
     embed = [":rttanalysis"],
     exec_properties = {"test.Pool": "large"},
+    shard_count = 4,
     deps = [
         "//pkg/base",
-        "//pkg/jobs",
         "//pkg/jobs/jobspb",
         "//pkg/security/securityassets",
         "//pkg/security/securitytest",
@@ -69,6 +71,7 @@ go_test(
         "//pkg/testutils/serverutils",
         "//pkg/testutils/skip",
         "//pkg/testutils/testcluster",
+        "//pkg/util/envutil",
         "//pkg/util/protoutil",
         "//pkg/util/randutil",
     ],

--- a/pkg/bench/rttanalysis/registry.go
+++ b/pkg/bench/rttanalysis/registry.go
@@ -9,6 +9,8 @@ import (
 	"strings"
 	"testing"
 
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
 	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/errors"
 	"github.com/stretchr/testify/require"
@@ -51,15 +53,66 @@ func (r *Registry) Run(b *testing.B) {
 // benchmarks can be filtered by passing the usual test filters underneath
 // this test's name.
 //
-// It takes a long time and thus is skipped under stress, race
-// and short.
+// It takes a long time and thus is skipped under duress and short.
 func (r *Registry) RunExpectations(t *testing.T) {
-	skip.UnderStress(t)
-	skip.UnderRace(t)
-	skip.UnderShort(t)
-	skip.UnderDeadlock(t)
+	r.RunExpectationsSharded(t, 1, 1)
+}
 
-	runBenchmarkExpectationTests(t, r)
+// RunExpectationsSharded runs all the benchmarks for one iteration
+// and validates that the number of RPCs meets the expectation. If run
+// with the --rewrite flag, it will rewrite the run benchmarks. The
+// benchmarks can be filtered by passing the usual test filters underneath
+// this test's name.
+//
+// It takes a long time and thus is skipped under duress and short.
+//
+// When shard and totalShards are provided (> 1), only a subset of benchmarks
+// assigned to the specific shard will be run, enabling parallel execution.
+// Test groups are distributed across shards using round-robin assignment.
+func (r *Registry) RunExpectationsSharded(t *testing.T, shard, totalShards int) {
+	defer jobs.TestingSetIDsToIgnore(map[jobspb.JobID]struct{}{3001: {}, 3002: {}})()
+	skip.UnderDuress(t)
+	skip.UnderShort(t)
+
+	// If totalShards is 1, run all tests; otherwise shard them
+	var registryToUse *Registry
+	if totalShards <= 1 {
+		// Run all test groups
+		registryToUse = r
+	} else {
+		// Create a registry with only the test groups assigned to this shard
+		shardRegistry := &Registry{
+			numNodes: r.numNodes,
+			cc:       r.cc,
+			r:        make(map[string][]RoundTripBenchTestCase),
+		}
+
+		// Distribute test groups across shards using round-robin assignment
+		// First, get all group names and sort them for consistent ordering
+		groupNames := make([]string, 0, len(r.r))
+		for groupName := range r.r {
+			groupNames = append(groupNames, groupName)
+		}
+		// Sort for deterministic assignment across runs
+		for i := 0; i < len(groupNames); i++ {
+			for j := i + 1; j < len(groupNames); j++ {
+				if groupNames[i] > groupNames[j] {
+					groupNames[i], groupNames[j] = groupNames[j], groupNames[i]
+				}
+			}
+		}
+
+		// Assign groups to shards using round-robin
+		for i, groupName := range groupNames {
+			assignedShard := (i % totalShards) + 1
+			if assignedShard == shard {
+				shardRegistry.r[groupName] = r.r[groupName]
+			}
+		}
+		registryToUse = shardRegistry
+	}
+
+	runBenchmarkExpectationTests(t, registryToUse)
 }
 
 // Register registers a set of test cases to a given benchmark name. It is

--- a/pkg/bench/rttanalysis/validate_benchmark_data_test.go
+++ b/pkg/bench/rttanalysis/validate_benchmark_data_test.go
@@ -6,13 +6,44 @@
 package rttanalysis
 
 import (
+	"strconv"
 	"testing"
 
-	"github.com/cockroachdb/cockroach/pkg/jobs"
-	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/util/envutil"
 )
 
-func TestBenchmarkExpectation(t *testing.T) {
-	defer jobs.TestingSetIDsToIgnore(map[jobspb.JobID]struct{}{3001: {}, 3002: {}})()
-	reg.RunExpectations(t)
+// NOTE: If you change the number of shards, you must also update the
+// shard_count in BUILD.bazel to match.
+const shardCount = 4
+
+// Validate that shardCount matches TEST_TOTAL_SHARDS environment variable at init time
+var _ = func() int {
+	totalShardsStr, found := envutil.ExternalEnvString("TEST_TOTAL_SHARDS", 1)
+	if totalShardsStr == "" || !found {
+		return 0
+	}
+	totalShards, err := strconv.Atoi(totalShardsStr)
+	if err != nil {
+		return 0
+	}
+	if totalShards != shardCount {
+		panic("shardCount mismatch: update shard_count in pkg/bench/rttanalysis/BUILD.bazel to match shardCount constant")
+	}
+	return 0
+}()
+
+func TestBenchmarkExpectationShard1(t *testing.T) {
+	reg.RunExpectationsSharded(t, 1, shardCount)
+}
+
+func TestBenchmarkExpectationShard2(t *testing.T) {
+	reg.RunExpectationsSharded(t, 2, shardCount)
+}
+
+func TestBenchmarkExpectationShard3(t *testing.T) {
+	reg.RunExpectationsSharded(t, 3, shardCount)
+}
+
+func TestBenchmarkExpectationShard4(t *testing.T) {
+	reg.RunExpectationsSharded(t, 4, shardCount)
 }

